cmd/argocd-server/commands/root.go

```diff
@@ -105,6 +107,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().IntVar(&listenPort, "port", common.DefaultPortAPIServer, "Listen on given port")
 	command.Flags().IntVar(&metricsPort, "metrics-port", common.DefaultPortArgoCDAPIServerMetrics, "Start metrics on given port")
 	command.Flags().IntVar(&repoServerTimeoutSeconds, "repo-server-timeout-seconds", 60, "Repo server RPC call timeout seconds.")
+	command.Flags().StringVar(&frameOptions, "x-frame-options", "sameorigin", "Set X-Frame-Options header in HTTP responses to `value`. To disable, set to \"\".")
```

Comment:
Should we default to off? I feel like there is risk of breaking existing installs.

Reply:
Hm, I think having sane default settings is important, especially regarding security. With this setting defaulting to off, we'd have users opt-in for security. I think if it breaks existing installs - which should only be rare cases, where people use iFrames to integrate ArgoCD into other applications from different domains. My opinion on this is the users should rather opt-out of the security feature by willingly disable it.

Reply:
I'm now wondering if the default should be `deny`?

Reply:
I think that depends on whether we use iFrames in the UI ourselfes - I didn't check that yet. If we do, default should stay at `samesite`, because `deny` would then certainly break stuff. 

But if we don't use iFrames at all - and don't plan to use them - probably `deny` would be a better default, yes.

Reply:
@alexmt

Reply:
I would use `samesite`. I cannot guarantee we never going to use iFrames.